src/banner/BannerProvider.jsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Alert from 'terra-alert';
+import ContentContainer from 'terra-content-container';
+
+import BannerRegistrationContext from './private/BannerRegistrationContext';
+import { organizeBannersByPriority } from './private/utils';
+
+const propTypes = {
+  /**
+   * Components to render within the context of the Banner Checkpoint. Any banners rendered within
+   * these child components will be prioritized and organized in a list by the Checkpoint and will be
+   * display above the children.
+   */
+  children: PropTypes.node,
+  /**
+   * By default, the children rendered by BannerProvider are fit to the Checkpoint's parent using 100% height.
+   * If `fitToParentIsDisabled` is provided, the Checkpoint will render at its intrinsic content height and
+   * potentially overflow its parent.
+   */
+  fitToParentIsDisabled: PropTypes.bool,
+};
+
+/**
+ * The Banner Checkpoint manages prioritizing and displaying all Workflow Banners
+ * rendered within the Checkpoint Context in a list above all other content in the tree.
+ */
+const BannerProvider = ({ fitToParentIsDisabled, children }) => {
+  const registeredBanners = React.useRef({});
+  const [banners, setBanners] = React.useState([]);
+
+  const registerBanner = (bannerId, bannerProps) => {
```

Comment:
Typically we've defined these APIs inside a useMemo block to prevent use from redefining these functions (and throwing them away) with every render.

```js
const bannerProviderValue = React.useMemo(() => {
  const registerBanner = ...;
  const unregisterBanner = ...;

  return {
    registerBanner,
    unregisterBanner,
  };
}, []);
```

Reply:
updated to use memo: https://github.com/cerner/terra-application/pull/45/commits/a905caaef9296be5ffa5857c8281de69a4c9595f#diff-51cafba97ae92a564c90ef1fa3bdbfafR32